### ISSUE TITLE
Continuous nonsense voice instructions

### ISSUE THREAD

**Issue by DamianoPantani** [osmandapp/OsmAnd#14337]
### Description
During navigation, on certain **straight streets with no intersections**, navigation continuously says `keep lept then in X meters keep left` (or right depending on your direction). It annoys the SH out of me especially when listening to music.

I've found the potential reason - it's when the road is super close to another line (not a road - boundary or a forest etc...), e.g.:

https://www.openstreetmap.org/#map=16/50.3239/20.3007
https://www.openstreetmap.org/#map=18/50.86121/20.17465

Map from 2nd link on a screenshot below, please note the route path has a big arrow indicating multiple turns, while it's a completely straight road:

![GX070029 MP4_snapshot_02 42 328](https://user-images.githubusercontent.com/16529024/169599486-2c7bcadd-299f-471e-aa60-da6115ccf13b.jpg)

### Your Environment
OsmAnd Version: `4.1.11` (2021-12-15 - newest as of 20th May 2022)
Android/iOS version: `Android 12`
Device model: `Samsung Galaxy S10`

Unsure if relevant but here is some debug info:
**Maps used:**  `offline`
**Voice used:**  `English (United Kingdom)`
**Traffic warnings:**  `Enabled`

**Comment by scaidermern** [comment]
Is this a regular navigation, a planned route or are you following an imported GPX track?

**Comment by DamianoPantaniHT** [comment]
@scaidermern it was a planned route, cannot remember where I put waypoints, but certainly there was at most 1 on this particular street

**Comment by vshcherb** [comment]
We would need GPX track if possible, probably as private message to test it out.

**Comment by Zirochkabila** [comment]
> https://drive.google.com/file/d/12xDo1rUKPiVbk5AS4cBDsNRWIHYAwW3r/view?usp=sharing

The case was checked on version 4.1.11 of OsmAnd - voice prompts work correctly.

Try to update the maps.

**Comment by scaidermern** [comment]
> I also had similar problem. It showed me similar instructions in the middle of highway (keep right, which does not have any sense). I have also lanes display enabled and it showed double the count of lanes as normal (showed 6 lanes when there were 3). I figured out later it was due to live updates (i could see after large zoom on those sections there were some doubled roads as they moved a bit in live update between some nodes). I've removed all updates and updated map (was like half year old) and the problem disappeared.

I had exactly the same problem with the same solution on OsmAnd 4.1.x. But this happened only once and never again.

**Comment by Zirochkabila** [comment]
> I see the same phenomenon. Is the information we have about the case enough, or is there any information I can add to the case to help in diagnosing the problem?

please add information

**Comment by Zirochkabila** [comment]
Your gpx track has been broken. 
Describe how you created it and made changes to it please

> GPX looked like follows: https://drive.google.com/file/d/12xDo1rUKPiVbk5AS4cBDsNRWIHYAwW3r/view?usp=sharing

**Comment by DamianoPantaniHT** [comment]
So I managed to reproduce with this exact GPX:
https://drive.google.com/file/d/12xDo1rUKPiVbk5AS4cBDsNRWIHYAwW3r/view

Location:
https://www.openstreetmap.org/#map=17/50.76935/20.45105
(heading south)

![GX030096 MP4_snapshot_01 34 679](https://user-images.githubusercontent.com/98806973/174482649-60967566-02c9-46a4-871f-b1e1f24bd2f2.png)

(again, notice that the blue path is highlighted with countless yellow turn arrows, merged into 1)

Debug info: I indeed have live updates turned on (for Poland -> Holy Cross province), I have a lifetime Maps+ license and I am an active OSM mapper

**Comment by Zirochkabila** [comment]
> So I managed to reproduce with this exact GPX: 

Your GPX track is still breaking.
Regarding the yellow navigation arrow - as you can see in the added screenshot, the route is displayed without the yellow arrow (live updates are enabled)

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/174571401-660812aa-8e56-4a0f-bc91-40d4ea9ffb92.jpeg"  width="300" height="auto"/> </th>
  </tr>
</table>  

What profile are you using when building your route (did you change something in profile settings) what is on the screenshot and GPX route?

**Comment by DamianoPantaniHT** [comment]
> What profile are you using when building your route (did you change something in profile settings) what is on the screenshot and GPX route?

I use a customized "Motorcycle" profile (there is a bunch of custom settings, not sure which are relevant):
https://drive.google.com/file/d/1UpvrxViniupg8O0Hsq0PexXdvlkg2_1J/view

There is no waypoint in this particular area on the GPX track:

![Screenshot_20220620-114955_OsmAnd](https://user-images.githubusercontent.com/98806973/174576447-882278c2-4137-4792-a275-02604788fde6.jpg)

> Please don't make photos while driving. Especially not on a motorcycle 🙈.

Bikers are built different, no worries about me 😋
(They've got GoPro mounted on helmets)

**Comment by Zirochkabila** [comment]
> https://www.openstreetmap.org/#map=17/50.76935/20.45105

The case with the yellow line on the route was checked with the profile settings that you sent - as you can see in the screenshot, everything works fine.
Try to update the application to a newer version - maybe that will solve the issue.

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/174658057-b0eedf71-f89a-4be9-bbdc-08390c31d839.jpeg"  width="300" height="auto"/> </th>
  </tr>
</table>  


If for this navigation (where the yellow line is displayed on the route) you use a GPX track - send it too please

**Comment by DamianoPantaniHT** [comment]
I use the newest version 4.2.4, and I already exported and shared GPX track (created in OsmAnd route planner)

Please note you won't be able to reproduce it by creating new route on this road only. Some specific preconditions must be met, but 0 idea what they are...

**Comment by DamianoPantaniHT** [comment]
So... I managed to reproduce with the GPX I shared, using OsmAnd "Simulate navigation" tool. Takes a while but eventually this happens:

[Video with no sound here](https://drive.google.com/file/d/197VipHV6Z2msPT6Cv7vFWczD9BHJJgzS/view) - but lady wanted me to "keep left then in X meters keep left" about 20 times. Please notice how the instruction in top left panel change every few meters

![Screenshot_20220620-212816_OsmAnd](https://user-images.githubusercontent.com/98806973/174668895-c86811c1-7f3b-483d-8466-3f9ccb252b98.jpg)

**Comment by Zirochkabila** [comment]
The GPX track you sent 
[Wed 25 May 2022.gpx.zip](https://github.com/osmandapp/OsmAnd/files/8950357/Wed.25.May.2022.gpx.zip)
 has nothing to do with the route on which the yellow turn line is displayed

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/174840539-fd0f0823-d0f8-4438-b944-40321f7c9d91.jpeg"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by DamianoPantaniHT** [comment]
You used wrong GPX file. Let me sum everything up here again so all is clear:

[GPX file (created in OsmAnd route planner)](https://drive.google.com/file/d/12xDo1rUKPiVbk5AS4cBDsNRWIHYAwW3r/view)
[Location (heading south)](https://www.openstreetmap.org/#map=17/50.76935/20.45105)
[Profile settings](https://drive.google.com/file/d/1UpvrxViniupg8O0Hsq0PexXdvlkg2_1J/view)
[Video - used navigation simulation to reproduce](https://drive.google.com/file/d/197VipHV6Z2msPT6Cv7vFWczD9BHJJgzS/view)

Please note you won't be able to reproduce it by creating new route on this road only. It must be this specific GPX

![Screenshot_20220620-212816_OsmAnd](https://github.com/assets/98806973/c86811c1-7f3b-483d-8466-3f9ccb252b98)

Debug info:
OsmAnd Version: **4.2.4**
Live updates: **enabled**
Device: **Samsung Galaxy S10 (Android)**

**Comment by Zirochkabila** [comment]
The GPX file that you sent is in a different part of the map and has nothing to do with the route in Tokarnia (the left turn is displayed with a continuous line)

If the bug with the direction arrow is related to the GPX track, and on your screenshot, you move on the GPX, then send the GPX where you moved. The
<img width="993" alt="Screenshot 2022-06-23 at 13 47 36" src="https://user-images.githubusercontent.com/104760013/175284021-d6174563-ad2b-4b4d-87af-018b599b6e85.png">
 case of the arrow without GPX could not be reproduced, as can be seen in the attached screenshot above.

**Comment by DamianoPantaniHT** [comment]
My bad, I used a wrong link. Edited last 2 messages. Please re-check

**Comment by DamianoPantaniHT** [comment]
After another app update, today map had similar problem with the arrows, but there were no annoying voice prompts:

![GX030101 MP4_snapshot_04 15 238](https://user-images.githubusercontent.com/98806973/175775915-49472fbe-929f-4f93-8e0c-f560ca8ff591.png)

Location: https://www.openstreetmap.org/#map=17/50.62590/20.35621
Please note that turn instructions in the top left corner are all correct (besides that I had 2000 m to go to roundabout, instead of 243 m, but that's a separate story).

I've seen a nice update on an expressway so that the app didn't tell me to "keep left" on every junction, and instead it just told me to continue for 65 km when I entered the road. Would that be related?

Considering above, should we close the issue? Or you want to confirm all is K?

**Comment by Zirochkabila** [comment]
1. When using your profile and gpx - the case repeats itself
2. When using your profile without gpx - the case does not repeat
3. When using the default profile without gpx - the case does not repeat
4. When using default profile and gpx - case repeats 

Conclusion - it is the GPX track, the arrow becomes continuous when using it

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/176214142-d8419ecc-2913-46a1-a94c-d49d9a5ee231.jpg"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/104760013/176214147-c9fd8845-2aca-49a7-9942-0bc1a77698b6.jpg"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/104760013/176214153-8c398cf6-50c9-4d92-8c84-5dee2e2124c5.jpg"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/104760013/176214158-60b67b21-9976-43d4-98e7-0f89fc979003.jpg"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by Zirochkabila** [comment]
When navigating on your own GPX, there are no solid navigation arrows 

[Wed 13 Jul 2022.gpx.zip](https://github.com/osmandapp/OsmAnd/files/9102345/Wed.13.Jul.2022.gpx.zip)

**Comment by DamianoPantani** [comment]
Today it reproduced 20+ times on this 260 km route:
https://drive.google.com/file/d/135qiX0yUdAREjSklyK10oHLgPvsPegk5/view

Example locations:
https://www.openstreetmap.org/#map=17/50.88396/20.09641 (km 30, heading W)
https://www.openstreetmap.org/#map=17/50.75229/19.30736 (km 136, heading E)
So many times between https://www.openstreetmap.org/#map=18/50.62668/19.81095 and https://www.openstreetmap.org/#map=17/50.67740/20.10187 especially closer to the end (on the picture km 209, heading E)

![image](https://user-images.githubusercontent.com/16529024/179360770-c890cf1e-0b4b-4282-a726-1f49bcd2881c.png)


On some locations I have my LiveUpdate on, and on some not. Recently though I downloaded fresh maps for entire Poland, may be related / relevant.

**Comment by zyphlar** [comment]
I've always gotten these repetitive nonsense alerts including today with latest OsmAnd and liveupdate maps.

My guess is that it has to do with "sausage roads" in many US cities: places where the road appears to diverge at a 45° angle only to rejoin a few feet later because there is a short median separating traffic. I don't want to give out my city of residence here but pick any medium sized US city and plan a route that involves some of these "sausage roads."

**Comment by Zirochkabila** [comment]
Case managed to reproduce, application version 4.2.7, released: 2022-07-13

https://user-images.githubusercontent.com/104760013/182398331-fe52ebe2-96f5-4b50-b61b-02cc46ed2c15.mp4

**Comment by ivanPyrohivskyi** [comment]
Must be fixed in commit https://github.com/osmandapp/OsmAnd/commit/3718236a78cbaf829dcead54ee600bc01270a6c2
